force-app/main/default/classes/DisplayPDFController.cls

```diff
@@ -258,37 +405,211 @@ global with sharing class DisplayPDFController
-    }
+    /**
+     * @description    Property to store PDF modes 
+     */
+    @AuraEnabled
+    public String pdfModes1 { get; set; }
     
+    /**
+     * @description Custom exception class used for handling exceptions in SafeClass.
+     */    
     public class SafeClassException extends Exception{}
     
-    public class headerfooterwrapper{
-        public String value {get; set;}
-        public String indexvar {get; set;}
+    /**
+     * @description Wrapper class for header and footer values
+     */
+    public class HeaderFooterWrapper {
+		/**
+         * @description Represents a value associated with a header or footer section.
+         */
+        public String value { get; set; } 
+        
+        /**
+         * @description Represents the index variable associated with a header or footer section.
+         */
+        public String indexvar { get; set; } 
         
-        public headerfooterwrapper(String value,String indexvar) {
+        /**
+         * @description Constructor for the HeaderFooterWrapper class.
+         * @param value The value associated with a header or footer section.
+         * @param indexvar The index variable associated with a header or footer section.
+         */
+        public HeaderFooterWrapper(String value, String indexvar) {
             this.value = value;
             this.indexvar = indexvar;
         }
     }
+    
+    /**
+     * @description    Retrieves a list of Opportunities created within the last 30 days
+     * @return         List<Opportunity> List of Opportunities
+     */
+    @AuraEnabled(cacheable=true)
+    public static List<Opportunity> getOpportunities() {
+        try {
+            List<Opportunity> opportunities = [SELECT Amount, Name, StageName 
+                                               FROM Opportunity 
+                                               WHERE CreatedDate = LAST_N_DAYS:30 WITH USER_MODE];
+            return opportunities;
+        } catch(Exception ex) {
+             LogHandler.createLog((Id) null,'DisplayPDFController.getOpportunities()','Template Id- '+ex.getLineNumber()+ex.getStackTraceString()+ex.getCause(),ex.getMessage(),'Exception');
+            throw new AuraHandledException(ex.getMessage());
+        }
+    }
+     
+    /**
+     * @description    Saves content versions (text or image) to Salesforce as ContentVersion records
+     * @param          title           Title of the content version
+     * @param          base64DataList List of maps containing base64-encoded data and record types (text or image)
+     * @param          templateId      Id of the document template where the content versions should be associated
+     * @return         List<Map<String, String>> List of maps containing saved record types and their corresponding Ids
+     */
+   @AuraEnabled(cacheable=false)
+public static List<Map<String, String>> saveContentVersion(String title, List<Map<String, String>> base64DataList, String templateId) {
```

Comment:
The Base64Data is a collection and Title is of a primitive data type. Conflict with the understanding.